app/callbacks/views/sns.py

```diff
@@ -217,220 +225,232 @@ def handle_s3_sns():
             message=lambda _: (
-                "Handled bucket {s3_bucket_name} key {s3_object_key} version {s3_object_version}"
+                "Scanned {file_length}byte file '{file_name}', result {clamd_result}"
                 if sys.exc_info()[0] is None else
-                # need to literally format() the exception into message as it's difficult to get it injected into extra
-                "Failed handling {{s3_bucket_name}} key {{s3_object_key}} version {{s3_object_version}}: {!r}".format(
+                # need to literally format() exception into message as it's difficult to get it injected into extra
+                "Failed scanning {{file_length}}byte file '{{file_name}}': {!r}".format(
                     sys.exc_info()[1]
                 )
             ),
             log_level=logging.INFO,
             condition=True,
-        ) as log_context:
-            s3_bucket_name = record["s3"]["bucket"]["name"]
-            s3_object_key = record["s3"]["object"]["key"]
-            s3_object_version = record["s3"]["object"]["versionId"]
-            base_log_context = {
-                "s3_bucket_name": s3_bucket_name,
-                "s3_object_key": s3_object_key,
-                "s3_object_version": s3_object_version,
-            }
-            log_context.update(base_log_context)
-
-            # TODO abort if file too big?
-
-            s3_client = boto3.client("s3", region_name=record["awsRegion"])
-
-            with log_external_request(
-                "S3",
-                "get object tagging [{s3_bucket_name}/{s3_object_key} versionId {s3_object_version}]",
-                logger=current_app.logger,
-            ) as log_context_s3:
-                log_context_s3.update(base_log_context)
-                tagging_tag_set = s3_client.get_object_tagging(
-                    Bucket=s3_bucket_name,
-                    Key=s3_object_key,
-                    VersionId=s3_object_version,
-                )["TagSet"]
-
-            av_status_json = _tag_value_from_tag_set(tagging_tag_set, "avStatus")
-
-            if av_status_json is None:
-                current_app.logger.info(
-                    "Object version {s3_object_version} has no 'avStatus' tag - will scan...",
-                    extra={
-                        **base_log_context,
-                        "av_status": av_status_json,
-                    },
+        ) as log_context_clamd:
+            log_context_clamd.update({
+                "file_length": s3_object["ContentLength"],
+                "file_name": file_name or "<unknown>",
+            })
+            clamd_result = clamd_client.instream(s3_object["Body"])["stream"]
+            log_context_clamd["clamd_result"] = clamd_result
+
+        if clamd_result[0] == "ERROR":
+            # let's hope this was a transient error and a later attempt may succeed. hard to know what else to do
+            # in this case - tagging a file with "ERROR" would prevent further attempts.
```

Comment:
We now only scan if it doesn't have `avStatus.result`. If there's an error, could we not tag it with something different so at least we know it's tried/failed? `avScan.failed` or something?

Reply:
We *could*, but I don't really know what these possible errors *are*. To my mind, here, tags are really so that automated processes can do something based on previous ... things ... that have happened. For us, wanting diagnostics, the logs should be enough I would have though.